server/api/socket/marker.py

```diff
@@ -0,0 +1,36 @@
+import auth
+from app import app, logger, sio, state
+from models import Marker
+from models.db import db
+
+
+@sio.on("Marker.New", namespace="/planarally")
+@auth.login_required(app, sio)
+async def new_marker(sid, data):
+    sid_data = state.sid_map[sid]
+    user = sid_data["user"]
+    location = sid_data["location"]
+
+    marker = Marker.get_or_none(shape=data, user=user)
+
+    if marker is not None:
+        return
+
+    Marker.create(
+        shape=data,
+        user=user,
+        location=location
+    )
+
+@sio.on("Marker.Remove", namespace="/planarally")
+@auth.login_required(app, sio)
+async def delete_marker(sid, uuid):
+    sid_data = state.sid_map[sid]
+    user = sid_data["user"]
+    location = sid_data["location"]
+
+    marker = Marker.get_or_none(shape_id=uuid)
```

Comment:
This one is not updated yet ?

Reply:
Yes, I did.... I noticed that `marker = Marker.get_or_none(uuid=uuid)` hadn't changed yet. Since the field name in the table is shape_id I changed it.

Reply:
No I mean it's missing user=user

Reply:
My bad, I changed the add-function instead of the delete-function.... sorted now.

Reply:
😄 It needs to be there in both cases

Reply:
otherwise other users won't be able to create a marker for a similar shape

Reply:
I know I shouldn't do two things at the same time.... 
Focus Leon, FOCUS!!!  ;-)